RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -160,14 +164,24 @@ extension PurchaseHandler {
     func purchase(package: Package) async throws {
         switch self.purchases.purchasesAreCompletedBy {
         case .revenueCat:
-            try await performPurchase(package: package)
+            try await performPurchase(package: package, promotionalOffer: nil)
+        case .myApp:
+            try await performExternalPurchaseLogic(package: package, promotionalOffer: nil)
+        }
+    }
```

Comment:
Nit: I would replace the whole method body with 
```swift
func purchase(package: Package) async throws {
    try await purchase(package: package, promotionalOffer: nil)
}
```

_Sent from the good old Objective-C days_